File: MomentsJava/src/main/java/com/example/momentsjava/data/repository/ListRepository.java
package com.example.momentsjava.data.repository;

import com.example.momentsjava.data.datasource.ListDataSource;
import com.example.momentsjava.data.model.ListItem;

import java.util.List;

import io.reactivex.rxjava3.core.Observable;

public class ListRepository {
    private final ListDataSource listDataSource;

    public ListRepository(ListDataSource listDataSource) {
        this.listDataSource = listDataSource;
    }

    public Observable<List<ListItem>> getList() {
        return listDataSource.getList();
    }
}

File: MomentsKotlin/src/main/java/com/example/moments/data/repository/ListRepository.kt
package com.example.moments.data.repository

import com.example.moments.data.datasource.ListDataSource
import com.example.moments.data.model.ListItem
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.runBlocking
import kotlinx.coroutines.withContext
import java.util.concurrent.ExecutorService
import java.util.concurrent.Executors

class ListRepository(private val listDataSource: ListDataSource) {
    private val executorService: ExecutorService = Executors.newFixedThreadPool(4)
    fun getList(): List<ListItem> {
            val res = executorService.submit<List<ListItem>> {
                runBlocking {
                    listDataSource.fetchList()
                }
            }
           return res.get()
    }
}
